bpf/kmesh/workload/include/authz.h

```diff
@@ -0,0 +1,213 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Kmesh */
+
+#ifndef __AUTHZ_H__
+#define __AUTHZ_H__
+
+#include "workload_common.h"
+#include "bpf_log.h"
+#include "xdp.h"
+#include "workloadapi/security/authorization.pb-c.h"
+
+#define AUTH_ALLOW  0
+#define AUTH_DENY   1
+#define UNMATCHED   0
+#define MATCHED     1
+#define UNSUPPORTED 2
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(key_size, sizeof(__u32));
+    __uint(value_size, sizeof(Istio__Security__Authorization));
+    __uint(map_flags, BPF_F_NO_PREALLOC);
+    __uint(max_entries, MAP_SIZE_OF_AUTH_POLICY);
+} map_of_authz SEC(".maps");
+
+static inline Istio__Security__Authorization *map_lookup_authz(__u32 policyKey)
+{
+    return (Istio__Security__Authorization *)kmesh_map_lookup_elem(&map_of_authz, &policyKey);
+}
+
+static inline wl_policies_v *get_workload_policies_by_uid(__u32 workload_uid)
+{
+    return (wl_policies_v *)kmesh_map_lookup_elem(&map_of_workload_policy, &workload_uid);
+}
+
+static inline int matchDstPorts(Istio__Security__Match *match, struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+{
+    __u32 *notPorts = NULL;
+    __u32 *ports = NULL;
+    __u32 i;
+
+    if (match->n_destination_ports == 0 && match->n_not_destination_ports == 0) {
+        return MATCHED;
+    }
+
+    if (match->n_not_destination_ports != 0) {
+        notPorts = kmesh_get_ptr_val(match->not_destination_ports);
+        if (!notPorts) {
+            return UNMATCHED;
+        }
+#pragma unroll
+        for (i = 0; i < MAX_MEMBER_NUM_PER_POLICY; i++) {
+            if (i >= match->n_not_destination_ports) {
+                break;
+            }
+            if (info->iph->version == 4) {
+                if (bpf_htons(notPorts[i]) == tuple_info->ipv4.dport) {
+                    return UNMATCHED;
+                }
+            } else {
+                if (bpf_htons(notPorts[i]) == tuple_info->ipv6.dport) {
+                    return UNMATCHED;
+                }
+            }
+        }
+    }
+
+    if (match->n_destination_ports != 0) {
+        ports = kmesh_get_ptr_val(match->destination_ports);
+        if (!ports) {
+            return UNMATCHED;
+        }
+#pragma unroll
+        for (i = 0; i < MAX_MEMBER_NUM_PER_POLICY; i++) {
+            if (i >= match->n_destination_ports) {
+                break;
+            }
+            if (info->iph->version == 4) {
+                if (bpf_htons(ports[i]) == tuple_info->ipv4.dport) {
+                    return MATCHED;
+                }
+            } else {
+                if (bpf_htons(ports[i]) == tuple_info->ipv6.dport) {
+                    return MATCHED;
+                }
+            }
+        }
+    }
+    return UNMATCHED;
```

Comment:
If the `not_description` rules are not matched, should it return matched?

Reply:
check the begin of this function， line 42 solved the scenario you mentioned

Reply:
fix